### PR TITLE
feat(flags): add support for feature flags

### PR DESCRIPTION
**Pull request by Maddiaa0**
# Description

Adds support to noirup to install from source or from a git repo with specific feature flags enabled.

Usage:
```bash
The installer for Nargo.
Update or revert to a specific Nargo version with ease.
USAGE:
    noirup <OPTIONS>
OPTIONS:
    -h, --help      Print help information
    -v, --version   Install a specific version
    -b, --branch    Install a specific branch
    -P, --pr        Install a specific Pull Request
    -C, --commit    Install a specific commit
    -r, --repo      Install from a remote GitHub repo (uses default branch if no other options are set)
    -p, --path      Install a local repository
    -f, --features  Activates feature flags when building from source
```

Example: 
If installing from a local repo with the aztec feature flag enabled:
```bash
noirup -p . -f aztec
```

## Problem\*

Resolves 

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: noirup [hunk 1]
```diff
@@ -16,6 +16,7 @@ main() {
       -r|--repo)        shift; NOIRUP_REPO=$1;;
       -b|--branch)      shift; NOIRUP_BRANCH=$1;;
       -v|--version)     shift; NOIRUP_VERSION=$1;;
+      -f|--features)     shift; NOIRUP_FEATURES=$1;;
```
Comment (TomAFrench) [inline noirup:19]:
```suggestion
      -f|--features)    shift; NOIRUP_FEATURES=$1;;
```

File: noirup [hunk 2]
```diff
@@ -48,7 +51,7 @@ main() {
     # Enter local repo and build
     say "installing from $NOIRUP_LOCAL_REPO"
     cd $NOIRUP_LOCAL_REPO
-    RUSTFLAGS="-C target-cpu=native" ensure cargo build --release # need 4 speed
+    RUSTFLAGS="-C target-cpu=native" ensure cargo build --release --features $FEATURES # need 4 speed
```
Comment (TomAFrench) [inline noirup:54]:
```suggestion
    RUSTFLAGS="-C target-cpu=native" ensure cargo build --release --no-default-features --features $FEATURES # need 4 speed
```

If we're allowing people to specify feature flags then we should allow them to opt out of the default ones.

File: noirup [hunk 3]
```diff
@@ -146,7 +149,7 @@ main() {
       ensure git checkout ${NOIRUP_COMMIT}
     fi
 
-    RUSTFLAGS="-C target-cpu=native" ensure cargo build --release
+    RUSTFLAGS="-C target-cpu=native" ensure cargo build --release --features $FEATURES
```
Comment (TomAFrench) [inline noirup:152]:
```suggestion
    RUSTFLAGS="-C target-cpu=native" ensure cargo build --release --no-default-features --features $FEATURES
```

File: noirup [hunk 4]
```diff
@@ -28,6 +29,8 @@ main() {
     esac; shift
   done
 
+  FEATURES=${NOIRUP_FEATURES:-"default"}
```
Comment (TomAFrench) [inline noirup:32]: Can you add a warning that features are ignored if we end up downloading a prebuilt binary?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -115,8 +115,22 @@ jobs:
       - run: nargo new project
       - run: nargo check
         working-directory: ./project
-      - run: nargo compile test-circuit
+      - run: |
```
Comment (TomAFrench) [inline .github/workflows/ci.yml:118]: oof

**Review by TomAFrench — COMMENTED** [review]
Couple of small changes but looks good.

**Comment by Savio-Sou** [comment]
@Maddiaa0 good to re-review?

**Comment by Maddiaa0** [comment]
I think so

**Review by TomAFrench — APPROVED** [review]
LGTM

**Pull request merged**